src/App.tsx

```diff
@@ -1,44 +1,101 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
-import './App.scss';
-
 import classNames from 'classnames';
+
+import './App.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './utils/users';
+import { getUserPosts } from './utils/posts';
+import { Post } from './types/Post';
+import { errorMessage } from './utils/errorMessage';
 
 export const App: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isPending, setIsPending] = useState(false);
+  const [isFail, setIsFail] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [areCommentsLoading, setAreCommentsLoading] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers);
+  }, []);
+
+  const handleClickOnUser = (clickedUser: User) => {
+    setUser(clickedUser);
+    setIsPending(true);
+    setSelectedPost(null);
+
+    getUserPosts(clickedUser.id)
+      .then(setPosts)
+      .catch((error) => {
+        setIsFail(true);
+        throw error;
```

Comment:
don't throw errors in catch brunch. it'll be unhandled and your script could fail